### PR TITLE
Remove redundant include getopt.h (QNX)

### PR DESCRIPTION
**Pull request by coypoop**
Aside from being redundant, it is absent in QNX 6.5, according to this report (not mine, I cannot run the testsuite):
https://mail-index.netbsd.org/pkgsrc-users/2017/09/20/msg025601.html

(getopt is in unistd.h)

### CODE REVIEW

**Comment by azat** [comment]
There are other places with `<getopt.h>`, can you fix them all?

```
sample/dns-example.c:#include <getopt.h>
test/bench.c:#include <getopt.h>
test/bench_cascade.c:#include <getopt.h>
```

BTW maybe you it is better to introduce `getopt_compat.h` that will:
```
#include <event2/event-config.h>

/* In QNX unistd.h includes getopt */
#ifdef EVENT__HAVE_UNISTD_H
#ifdef __QNX__
#define __EXT_POSIX2
#endif /* __QNX__ */
#include <unistd.h>
#endif /* EVENT__HAVE_UNISTD_H */

#ifndef __QNX__
#include <getopt.h>
#endif /* !__QNX__ */
```

**Comment by azat** [comment]
(P.S. And I didn't tested that chunk anyhow)

**Comment by coypoop** [comment]
After looking at the other cases, I think I know why it was necessary to include getopt.h:
searching says windows doesn't normally have getopt, and I think you have a shim thing to provide getopt on windows by including getopt.h (WIN32-Code/getopt.h)

In those other cases it is doing:

    #ifdef _WIN32
    #include <getopt.h>

so this would break Windows

**Comment by coypoop** [comment]
would it be okay if I provide a different pull request to:
provide getopt under unistd.h rather than getopt.h
change headers so win32 uses unistd.h (currently it's ifndef WIN32 everywhere)
never include getopt.h

I assume appveyor is testing the windows build (I don't have windows available)

**Comment by azat** [comment]
> #ifdef _WIN32
> #include <getopt.h>
>
> so this would break Windows

Yep, this will break win32

**Comment by azat** [comment]
> would it be okay if I provide a different pull request to:

Sure, go ahead!

> I assume appveyor is testing the windows build (I don't have windows available)

Yep, but it is a bit out-of-date.

We have testing in vagrant, that includes win32 box, so if you can ran
them, that would be great.
Here is a wrapper for them:
  https://github.com/libevent/libevent-extras/blob/master/tools/vagrant-tests.py

File: WIN32-Code/nmake/event2/event-config.h [hunk 1]
```diff
@@ -265,7 +265,7 @@
 /* #define EVENT__HAVE_UINT8_T 1 */
 
 /* Define to 1 if you have the <unistd.h> header file. */
-/* #define EVENT__HAVE_UNISTD_H 1 */
+#define EVENT__HAVE_UNISTD_H 1
```
Comment (azat) [inline WIN32-Code/nmake/event2/event-config.h]: Hm, but this does not fool cmake, it also should be modified.

**Comment by azat** [comment]
Something like referenced (fb42e0fa) patch?

**Comment by azat** [comment]
I wrapped that define under `#ifdef __QNX__` 99a3887d66885204ed8fa13405ae04b632dc8d33, objections?

**Comment by coypoop** [comment]
No, thanks!

**Pull request closed without merging**